pkg/registry/core/service/allocator/storage/storage.go

```diff
@@ -61,7 +62,12 @@ var _ rangeallocation.RangeRegistry = &Etcd{}
 // NewEtcd returns an allocator that is backed by Etcd and can manage
 // persisting the snapshot state of allocation after each allocation is made.
 func NewEtcd(alloc allocator.Snapshottable, baseKey string, resource schema.GroupResource, config *storagebackend.Config) *Etcd {
-	storage, _ := generic.NewRawStorage(config)
+	storage, d := generic.NewRawStorage(config)
+	destroyFunc := func() {
```

Comment:
why this closure? d should work directly.

Reply:
y, cleaning up